redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/RealTreehouseApp.kt

```diff
@@ -192,7 +191,11 @@ internal class RealTreehouseApp<A : AppService> private constructor(
     ): TreehouseApp<A> = RealTreehouseApp(this, appScope, spec, eventListenerFactory)
 
     override fun close() {
-      cache.close()
+      if (cache.isInitialized()) {
+        cache.value.close()
+      }
```

Comment:
Otherwise, we'd hit `error("unexpected call")` @ https://github.com/cashapp/redwood/blob/fc483a2de5530bf5992c62d50170bf150886c08c/redwood-treehouse-host/src/appsJvmTest/kotlin/app/cash/redwood/treehouse/TreehouseTester.kt#L72-L74.